core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -472,8 +472,11 @@ public void updateShardState(final ShardRouting newRouting,
                                  * subsequently fails before the primary/replica re-sync completes successfully and we are now being
                                  * promoted, the local checkpoint tracker here could be left in a state where it would re-issue sequence
                                  * numbers. To ensure that this is not the case, we restore the state of the local checkpoint tracker by
-                                 * replaying the translog and marking any operations there are completed.
+                                 * replaying the translog and marking any operations there are completed. Rolling the translog generation is
+                                 * not strictly needed here, but it simplifies reasoning about the relationship between primary terms and
```

Comment:
can we add to "not strictly needed here" , "not strictly needed here (we will never have collisions between sequences numbers in the translog files in a new primary as it takes the last know seq# as a starting point)"